core/src/main/scala/kafka/zookeeper/ZookeeperClient.scala

```diff
@@ -279,33 +258,130 @@ trait ZNodeChildChangeHandler {
+      override def processResult(rc: Int, path: String, ctx: Any): Unit =
+        processResponse(DeleteResponse(Code.get(rc), path, Option(ctx)))
+    }, ctx.orNull)
+  }
+}
+
+case class ExistsRequest(path: String, ctx: Option[Any] = None) extends WatchableAsyncRequest {
+  type Response = ExistsResponse
+
+  protected[zookeeper] def send(zooKeeper: ZooKeeper, watch: Boolean)(processResponse: Response => Unit): Unit = {
+    zooKeeper.exists(path, watch, new StatCallback {
+      override def processResult(rc: Int, path: String, ctx: Any, stat: Stat) =
+        processResponse(ExistsResponse(Code.get(rc), path, Option(ctx), stat))
+    }, ctx.orNull)
+  }
+}
+
+case class GetDataRequest(path: String, shouldWatch: Option[Boolean] = None, ctx: Option[Any] = None) extends WatchableAsyncRequest {
+  type Response = GetDataResponse
+
+  protected[zookeeper] def send(zooKeeper: ZooKeeper, watch: Boolean)(processResponse: Response => Unit): Unit = {
+    zooKeeper.getData(path, watch, new DataCallback {
+      override def processResult(rc: Int, path: String, ctx: Any, data: Array[Byte], stat: Stat) =
+        processResponse(GetDataResponse(Code.get(rc), path, Option(ctx), data, stat))
+    }, ctx.orNull)
+  }
+}
+
+case class SetDataRequest(path: String, data: Array[Byte], version: Int, ctx: Option[Any] = None) extends AsyncRequest {
+  type Response = SetDataResponse
+
+  protected[zookeeper] def send(zooKeeper: ZooKeeper)(processResponse: Response => Unit): Unit = {
+    zooKeeper.setData(path, data, version, new StatCallback {
+      override def processResult(rc: Int, path: String, ctx: Any, stat: Stat) =
+        processResponse(SetDataResponse(Code.get(rc), path, Option(ctx), stat))
+    }, ctx.orNull)
+  }
+}
+
+case class GetAclRequest(path: String, ctx: Option[Any] = None) extends AsyncRequest {
+  type Response = GetAclResponse
+
+  protected[zookeeper] def send(zooKeeper: ZooKeeper)(processResponse: Response => Unit): Unit = {
+    zooKeeper.getACL(path, null, new ACLCallback {
+      override def processResult(rc: Int, path: String, ctx: Any, acl: java.util.List[ACL], stat: Stat): Unit = {
+        processResponse(GetAclResponse(Code.get(rc), path, Option(ctx), Option(acl).map(_.asScala).getOrElse(Seq.empty),
+          stat))
+      }}, ctx.orNull)
+  }
+}
+
+case class SetAclRequest(path: String, acl: Seq[ACL], version: Int, ctx: Option[Any] = None) extends AsyncRequest {
+  type Response = SetAclResponse
+
+  protected[zookeeper] def send(zooKeeper: ZooKeeper)(processResponse: Response => Unit): Unit = {
+    zooKeeper.setACL(path, acl.asJava, version, new StatCallback {
+      override def processResult(rc: Int, path: String, ctx: Any, stat: Stat) =
+        processResponse(SetAclResponse(Code.get(rc), path, Option(ctx), stat))
+      }, ctx.orNull)
+  }
+}
+
+case class GetChildrenRequest(path: String, shouldWatch: Option[Boolean] = None, ctx: Option[Any] = None) extends WatchableAsyncRequest {
```

Comment:
Looks like the earlier refactoring was incomplete. `shouldWatch` shouldn't exist in the constructor.